Builtin append with only one argument fails

### The following program `sample.go` triggers an unexpected result

```go
package main

func main() {
	var C []int
	_ = append(C)
}
```


### Expected result

```console
$ go run ./sample.go
<no error>
```


### Got

```console
$ yaegi ./sample.go
run: ./main.go:5:6: CFG post-order panic: runtime error: index out of range [1] with length 1
```


### Yaegi Version

v0.9.23

### Additional Notes

In addition to the above error `append(nil)` fails with a nil-dereference error where the go compiler gives a suitable compiler error.

--- a/interp/interp_eval_test.go
+++ b/interp/interp_eval_test.go
@@ -496,6 +496,7 @@ func TestEvalSliceExpression(t *testing.T) {
 		{src: `a := []int{0,1,2,3}[1:3:]`, err: "1:51: 3rd index required in 3-index slice"},
 		{src: `a := []int{0,1,2}[3:1]`, err: "invalid index values, must be low <= high <= max"},
 		{src: `a := []int{0,1,2}; append(a)`, res: "[0 1 2]"},
+		{src: `append(nil)`, err: "first argument to append must be slice; have nil"},
 		{pre: func() { eval(t, i, `type Str = string; var r Str = "truc"`) }, src: `r[1]`, res: "114"},
 	})
 }

--- a/interp/interp_eval_test.go
+++ b/interp/interp_eval_test.go
@@ -140,6 +140,8 @@ func TestEvalBuiltin(t *testing.T) {
 		{src: `string(append([]byte("hello "), "world"...))`, res: "hello world"},
 		{src: `e := "world"; string(append([]byte("hello "), e...))`, res: "hello world"},
 		{src: `b := []int{1}; b = append(1, 2, 3); b`, err: "1:54: first argument to append must be slice; have int"},
+		{src: `a1 := []int{0,1,2}; append(a1)`, res: "[0 1 2]"},
+		{src: `append(nil)`, err: "first argument to append must be slice; have nil"},
 		{src: `g := len(a)`, res: "1"},
 		{src: `g := cap(a)`, res: "1"},
 		{src: `g := len("test")`, res: "4"},
@@ -495,8 +497,6 @@ func TestEvalSliceExpression(t *testing.T) {
 		{src: `a := []int{0,1,2,3}[1::4]`, err: "1:49: 2nd index required in 3-index slice"},
 		{src: `a := []int{0,1,2,3}[1:3:]`, err: "1:51: 3rd index required in 3-index slice"},
 		{src: `a := []int{0,1,2}[3:1]`, err: "invalid index values, must be low <= high <= max"},
-		{src: `a := []int{0,1,2}; append(a)`, res: "[0 1 2]"},
-		{src: `append(nil)`, err: "first argument to append must be slice; have nil"},
 		{pre: func() { eval(t, i, `type Str = string; var r Str = "truc"`) }, src: `r[1]`, res: "114"},
 	})
 }

--- a/interp/interp_eval_test.go
+++ b/interp/interp_eval_test.go
@@ -495,6 +495,7 @@ func TestEvalSliceExpression(t *testing.T) {
 		{src: `a := []int{0,1,2,3}[1::4]`, err: "1:49: 2nd index required in 3-index slice"},
 		{src: `a := []int{0,1,2,3}[1:3:]`, err: "1:51: 3rd index required in 3-index slice"},
 		{src: `a := []int{0,1,2}[3:1]`, err: "invalid index values, must be low <= high <= max"},
+		{src: `a := []int{0,1,2}; append(a)`, res: "[0 1 2]"},
 		{pre: func() { eval(t, i, `type Str = string; var r Str = "truc"`) }, src: `r[1]`, res: "114"},
 	})
 }

--- a/interp/run.go
+++ b/interp/run.go
@@ -2928,7 +2928,13 @@ func _append(n *node) {
 	value := genValue(n.child[1])
 	next := getExec(n.tnext)
 
-	if len(n.child) > 3 {
+	switch l := len(n.child); {
+	case l == 2:
+		n.exec = func(f *frame) bltn {
+			dest(f).Set(value(f))
+			return next
+		}
+	case l > 3:
 		args := n.child[2:]
 		l := len(args)
 		values := make([]func(*frame) reflect.Value, l)
@@ -2957,7 +2963,7 @@ func _append(n *node) {
 			dest(f).Set(reflect.Append(value(f), sl...))
 			return next
 		}
-	} else {
+	default:
 		var value0 func(*frame) reflect.Value
 		switch elem := n.typ.elem(); {
 		case isEmptyInterface(elem):

--- a/interp/typecheck.go
+++ b/interp/typecheck.go
@@ -739,6 +739,9 @@ func (check typecheck) builtin(name string, n *node, child []*node, ellipsis boo
 			return params[0].nod.cfgErrorf("first argument to append must be slice; have %s", typ.id())
 		}
 
+		if nparams == 1 {
+			return nil
+		}
 		// Special case append([]byte, "test"...) is allowed.
 		t1 := params[1].Type()
 		if nparams == 2 && ellipsis && t.Elem().Kind() == reflect.Uint8 && t1.TypeOf().Kind() == reflect.String {

--- a/interp/typecheck.go
+++ b/interp/typecheck.go
@@ -735,7 +735,7 @@ func (check typecheck) builtin(name string, n *node, child []*node, ellipsis boo
 	case bltnAppend:
 		typ := params[0].Type()
 		t := typ.TypeOf()
-		if t.Kind() != reflect.Slice {
+		if t == nil || t.Kind() != reflect.Slice {
 			return params[0].nod.cfgErrorf("first argument to append must be slice; have %s", typ.id())
 		}
 

